### ISSUE TITLE
File locking in Publish Assets job

### ISSUE THREAD

**Issue by missymessa** [dotnet/arcade#5491]
- [ ] This issue is blocking 
- [ ] This issue is causing unreasonable pain

dotnet/tye builds are experiencing this error

Link to build: https://dev.azure.com/dnceng/internal/_build/results?buildId=646386&view=logs&j=0b850c37-f3ae-5bb9-a9fb-4f1e0fce2747&t=50d41e72-ce5e-50a3-efc7-c28c030be91c

```
##[error].packages\microsoft.dotnet.arcade.sdk\5.0.0-beta.20228.4\tools\SdkTasks\PublishArtifactsInManifest.proj(79,5): error : (NETCORE_ENGINEERING_TELEMETRY=Publish) Unable to upload to assets/symbols/Microsoft.Tye.Core.0.3.0-alpha.20264.1.symbols.nupkg due to System.IO.IOException: The process cannot access the file 'd:\a\1\a\BlobArtifacts\Microsoft.Tye.Core.0.3.0-alpha.20264.1.symbols.nupkg' because it is being used by another process.
```

**Comment by riarenas** [comment]
I think we've seen some other file locking during publishing, and it was caused by multiple legs generating a manifest with the same set of files. Worth investigating if this is the case for dotnet/tye.

**Comment by jkotalik** [comment]
How would we go about investigating this? Is there a manifest I can look at which contains the set of files that will be published?

**Comment by JohnTortugo** [comment]
I think @riarenas is right here. @jkotalik - Look at the AzDO Artifacts "AssetManifests" folder. There are two xml files there and they list the same build assets. Do you generate assets with same name+version in different build legs?

**Comment by missymessa** [comment]
Requesting the RCA so the test gap can be identified for Arcade Validation.

**Comment by jkotalik** [comment]
I see, I doubt we need both those xml files, just one.

How would I go about fixing this? I don't see anything in particular that would be causing us to publish the same artifacts twice.

Yaml for reference: https://github.com/dotnet/tye/blob/master/azure-pipelines.yml#L58

**Comment by JohnTortugo** [comment]
@jkotalik - I might be wrong but I think you just need to pass `/p:Publish=false` when calling cibuild.sh here : https://github.com/dotnet/tye/blob/master/azure-pipelines.yml#L118

**Comment by jkotalik** [comment]
@sebastienros already fixed this via https://github.com/dotnet/tye/commit/ea864769586ef74caa88c8e580de5588b7d09fa2. My guess is that it accomplishes the same thing.

**Comment by jkotalik** [comment]
I went ahead and closed the automated RCA issue and removed RCA requested. Feel free to change if necessary.

**Comment by missymessa** [comment]
Thanks for letting me know, @jkotalik! If it was something on our end, I wanted to make sure we get additional test coverage for that.

**Comment by JohnTortugo** [comment]
> @sebastienros already fixed this via [dotnet/tye@ea86476](https://github.com/dotnet/tye/commit/ea864769586ef74caa88c8e580de5588b7d09fa2). My guess is that it accomplishes the same thing.

@jkotalik - Yeah, that's one way of doing it.
